src/Common/TransformEndianness.hpp

```diff
@@ -0,0 +1,54 @@
+#pragma once
+
+#include <base/Decimal_fwd.h>
+#include <base/extended_types.h>
+
+#include <utility>
+
+namespace DB
+{
+template <std::endian endian, typename T>
+    requires is_big_int_v<T>
```

Comment:
(cosmetic: let's not intend `requires`)

Reply:
I actually would've preferred it that way, but `clang-format` formatted it this way.

Reply:
If this is indeed desirable then I'll set https://clang.llvm.org/docs/ClangFormatStyleOptions.html#indentrequiresclause in the `.clang-format` file.